tensorboard/webapp/metrics/views/card_renderer/scalar_card_container.ts

```diff
@@ -611,10 +610,38 @@ export class ScalarCardContainer implements CardRenderer, OnInit, OnDestroy {
   onTimeSelectionChanged(
     newTimeSelectionWithAffordance: TimeSelectionWithAffordance
   ) {
-    this.store.dispatch(timeSelectionChanged(newTimeSelectionWithAffordance));
-    this.stepSelectorTimeSelection$.next(
-      newTimeSelectionWithAffordance.timeSelection
+    let newStart = newTimeSelectionWithAffordance.timeSelection.start;
+    const {minStep, maxStep} = this.minMaxSteps$.getValue();
+    newStart = {
+      step:
+        newStart.step < minStep
+          ? minStep
+          : newStart.step > maxStep
+          ? maxStep
+          : newStart.step,
+    };
+
+    let newEnd = newTimeSelectionWithAffordance.timeSelection.end;
+    if (newEnd !== null) {
+      newEnd = {
+        step:
+          newEnd.step < minStep
+            ? minStep
+            : newEnd.step > maxStep
+            ? maxStep
+            : newEnd.step,
+      };
+    }
+
+    const newTimeSelection = {start: newStart, end: newEnd};
```

Comment:
Maybe it's just me but the two sets of nested ternaries are pretty hard to read.

```suggestion
    const {minStep, maxStep} = this.minMaxSteps$.getValue();
    function bound(value: number) {
      if (value > maxStep) return maxStep;
      if (value < minStep) return minStep;
      return value;
    }
    const start = bound(newTimeSelectionWithAffordance.timeSelection.start.step);
    const end = newTimeSelectionWithAffordance.timeSelection.end ? 
                   bound(newTimeSelectionWithAffordance.timeSelection.end.step) :
                   null;
    const newTimeSelection = {start, end};
```